### PR TITLE
Adding PubSub v1 spec

### PR DESCRIPTION
**Pull request by nachocano**
Fixes https://github.com/google/knative-gcp/issues/544

## Proposed Changes

- 🎁Added the v1 PubSub binding

**Release Note**

```release-note
CloudEvents PubSub binding spec is now available
```

### CODE REVIEW

**Comment by nachocano** [comment]
/assign @grantr 

per CE folks [feedback](https://github.com/cloudevents/spec/pull/587#issuecomment-598234431), we should add it to our repo and then link it from there...

**Comment by nachocano** [comment]
@grantr ready for another look

**Comment by nachocano** [comment]
fyi @n3wscott

File: docs/spec/pubsub-protocol-binding.md [hunk 1]
```diff
@@ -153,7 +156,7 @@ Examples:
     * `id` maps to `ce-id`
     * `specversion` maps to `ce-specversion`
 
-Attribute names are case-insensitive.
+Attribute names are lower-case.
```
Comment (grantr) [inline docs/spec/pubsub-protocol-binding.md]: If all attribute names are lower case, it seems a bit odd to use `contentType` as the only attribute with an uppercase character.

**Comment by nachocano** [comment]
🤦🏻‍♂️ My bad, let's do content-type then, which seems to match binary
better

**Comment by grantr** [comment]
The `contentType` change came from https://github.com/cloudevents/spec/pull/587#discussion_r391700876, where it was suggested alongside `content-type`. I think `content-type` makes sense.

**Comment by nachocano** [comment]
fyi @nlopezgi for when you start with the sdk refactor. Some things changed, such as `Content-Type` for `content-type` and lower-casing. We can sync later on and I can show you where that is in the code.

^^^ @hrasadi -> this is the latest version so far

**Comment by nachocano** [comment]
/hold cancel
let's get this one in! need one more lgtm @grantr

File: docs/spec/pubsub-protocol-binding.md (hunk 1 above)
Comment (nachocano) [reply docs/spec/pubsub-protocol-binding.md]: changed it!

**Pull request merged**